src/gt4py/next/iterator/embedded.py

```diff
@@ -470,7 +472,7 @@ def promote_scalars(val: CompositeOfScalarOrField):
     elif isinstance(val, LocatedField):
         return val
     val_type = infer_dtype_like_type(val)
-    if np.issubdtype(val_type, np.number):
+    if isinstance(val, Scalar):  # type: ignore # mypy bug
```

Comment:
Please add a reference to the bug/issue description or a very short description of what's the problem here for mypy, otherwise the comment is not very useful.